### PR TITLE
CanBeSent func return true if the previous ConditionServiceLogSent status is false

### PR DESCRIPTION
**Pull request by a7vicky**
### What type of PR is this?
_(bug)_

### What this PR does / why we need it?
CanBeSent func return true if the previous ConditionServiceLogSent status is false

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_[OSD-20514](https://issues.redhat.com/browse/OSD-20514)

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: api/v1alpha1/managednotification_types.go [hunk 1]
```diff
@@ -212,8 +212,10 @@ func (m *ManagedNotification) CanBeSent(n string, firing bool) (bool, error) {
 		}
 		now := time.Now()
 		nextresend := sentCondition.LastTransitionTime.Time.Add(time.Duration(t.ResendWait) * time.Hour)
-		if now.Before(nextresend) {
+		if now.Before(nextresend) && sentCondition.Status == corev1.ConditionTrue {
 			return false, nil
+		} else {
+			return true, nil
```
Comment (ravitri) [inline api/v1alpha1/managednotification_types.go:218]: Wouldn't line 242 i.e https://github.com/openshift/ocm-agent-operator/pull/142/files#diff-327a7c4aef0339f4df5bbd9d48c90c5fa153639c504b9a49f22a24f2c7376f8cL242 in this PR already handle the `return true, nil`  after the if block for firing alert is covered? Let me know if I missed any context/intent here.

Comment (a7vicky) [reply api/v1alpha1/managednotification_types.go:218]: We want to return (true, nil) if the alert is firing and resend wait time has not been reached but the last SL sent condition status is false.

It'll not reach to line 242 for any firing alert which we want to cover in this.

Comment (a7vicky) [reply api/v1alpha1/managednotification_types.go:218]: Agree, It will cover, I think we can remove else statement but adding this also make it easier ?

Comment (ravitri) [reply api/v1alpha1/managednotification_types.go:218]: Well, in other words someone else might also consider it easier to read it as "in the code block do something **only if** condition is true". The "else" is implied which is outside the if block. 

Overall, the better reference for this query is https://go.dev/doc/effective_go#if 

> In the Go libraries, you'll find that when an if statement doesn't flow into the next statement—that is, the body ends in break, continue, goto, or return—the unnecessary else is omitted.

```go
f, err := os.Open(name)
if err != nil {
    return err
}
codeUsing(f)
```

Comment (a7vicky) [reply api/v1alpha1/managednotification_types.go]: @ravitri Thanks for the review, I have made the necessary changes you indicated

**Comment by a7vicky** [comment]
/tide/merge-method-squash

**Pull request merged**